### PR TITLE
AF-161 Fix Build

### PR DESCRIPTION
**Pull request by tleef42**
This is a PR to run and debug GitHub workflows 

Current changes:
- Add CLI publishing to make sure the JSONs end up published before attempting to validate them

### CODE REVIEW

File: src/attack_flow_builder/src/assets/builder.config.validator.ts [hunk 1]
```diff
@@ -203,11 +213,10 @@ class AttackFlowValidator extends DiagramValidator {
...
                 }
                 // Validate Payload Bin, URL, and Hashes
-                if(payloadBin?.isDefined() && url?.isDefined()) {
-                    this.addError(id, "Artifact must have either have a Payload Bin or URL, not both.");
-                }
-                if(url?.isDefined() && !hashes?.isDefined()) {
-                    this.addError(id, "Artifact URL must also have a Hash.");
+                if (payloadBin?.isDefined() && (url?.isDefined() || hashes?.isDefined()) ||
+                   (url?.isDefined() && !hashes?.isDefined()) ||
+                   (hashes?.isDefined() && !url?.isDefined())) {
```
Comment (mikecarenzo) [inline src/attack_flow_builder/src/assets/builder.config.validator.ts:218]: The [STIX Documentation](https://docs.oasis-open.org/cti/stix/v2.1/os/stix-v2.1-os.html#_4jegwl6ojbes) states that if `url` is present, `hashes` must be present. It never states that if `hashes` is present, `url` must be present.

File: src/attack_flow_builder/src/assets/builder.config.validator.ts [hunk 2]
```diff
@@ -203,11 +213,10 @@ class AttackFlowValidator extends DiagramValidator {
                     this.addError(id, "Invalid MIME Type.");
                 }
                 // Validate Payload Bin, URL, and Hashes
-                if(payloadBin?.isDefined() && url?.isDefined()) {
-                    this.addError(id, "Artifact must have either have a Payload Bin or URL, not both.");
-                }
-                if(url?.isDefined() && !hashes?.isDefined()) {
-                    this.addError(id, "Artifact URL must also have a Hash.");
+                if (payloadBin?.isDefined() && (url?.isDefined() || hashes?.isDefined()) ||
```
Comment (mikecarenzo) [inline src/attack_flow_builder/src/assets/builder.config.validator.ts:216]: The [STIX Documentation](https://docs.oasis-open.org/cti/stix/v2.1/os/stix-v2.1-os.html#_4jegwl6ojbes) states that `hashes`:
> Specifies a dictionary of hashes for the contents of the **url** or the **payload_bin**.

This `if` statement doesn't allow `payload_bin` and `hashes` to be defined at the same time.

**Review by mikecarenzo — CHANGES_REQUESTED** [review]

File: src/attack_flow_builder/src/assets/builder.config.validator.ts (hunk 1 above)
Comment (mehaase) [reply src/attack_flow_builder/src/assets/builder.config.validator.ts:218]: Ahh yeah I misread that. I'll fix it.

File: src/attack_flow_builder/src/assets/builder.config.validator.ts [hunk 3]
```diff
@@ -213,10 +213,10 @@ class AttackFlowValidator extends DiagramValidator {
                     this.addError(id, "Invalid MIME Type.");
                 }
                 // Validate Payload Bin, URL, and Hashes
-                if (payloadBin?.isDefined() && (url?.isDefined() || hashes?.isDefined()) ||
-                   (url?.isDefined() && !hashes?.isDefined()) ||
-                   (hashes?.isDefined() && !url?.isDefined())) {
-                        this.addError(id, "An artifact must have either a payload or a URL + hash, but not both.");
+                if (payloadBin?.isDefined() === url?.isDefined()) {
```
Comment (mikecarenzo) [inline src/attack_flow_builder/src/assets/builder.config.validator.ts]: This `if` statement fails if both `payload_bin` and `url` are blank, but according to the [STIX Documentation](https://docs.oasis-open.org/cti/stix/v2.1/os/stix-v2.1-os.html#_4jegwl6ojbes), both fields are optional.

**Review by mikecarenzo — APPROVED** [review]

File: src/attack_flow_builder/src/assets/builder.config.validator.ts (hunk 3 above)
Comment (mikecarenzo) [reply src/attack_flow_builder/src/assets/builder.config.validator.ts]: Fix applied.

**Pull request merged**